spec.html

```diff
@@ -20558,8 +20537,7 @@ <h1>Runtime Semantics: Evaluation</h1>
             1. Let _done_ be ? IteratorComplete(_innerResult_).
             1. If _done_ is *true*, then
               1. Return ? IteratorValue(_innerResult_).
-            1. If _generatorKind_ is ~async~, then set _received_ to AsyncGeneratorYield(? IteratorValue(_innerResult_)).
-            1. Else, set _received_ to GeneratorYield(_innerResult_).
+            1. Set _received_ to ? Yield(IteratorValue(_innerResult_)).
```

Comment:
This change causes the yielded inner result to no longer be passed through with synchronous generators:

```js
const result = {
	done: false,
	value: 'foo',
};

const iterable = {
	[Symbol.iterator]() {
		return iterable;
	},
	next() {
		return result;
	},
};

const gen = (function* () {
	return yield* iterable;
})();

assert.sameValue(gen.next(), result);
```

Reply:
probably the `?` is needed before both?

Reply:
Oh, I notice the difference, interesting. Why it is designed like this?